erizo_controller/erizoClient/src/ConnectionManager.js

```diff
@@ -0,0 +1,161 @@
+
+  that.erizoId = erizoId;
+  const streamsMap = ErizoMap(); // key:streamId, value: stream
+
+  const spec = specInput;
+  ErizoSessionId += 1;
+  spec.sessionId = ErizoSessionId;
+  that.sessionId = ErizoSessionId;
+
+  // Check which WebRTC Stack is installed.
+  that.browser = ConnectionHelpers.getBrowser();
+  if (that.browser === 'fake') {
+    Logger.warning('Publish/subscribe video/audio streams not supported in erizofc yet');
+    stack = FcStack(spec);
+  } else if (that.browser === 'mozilla') {
+    Logger.debug('Firefox Stack');
+    stack = FirefoxStack(spec);
+  } else if (that.browser === 'safari') {
+    Logger.debug('Safari using Chrome Stable Stack');
+    stack = ChromeStableStack(spec);
+  } else if (that.browser === 'chrome-stable' || that.browser === 'electron') {
+    Logger.debug('Chrome Stable Stack');
+    stack = ChromeStableStack(spec);
+  } else {
+    Logger.error('No stack available for this browser');
+    throw new Error('WebRTC stack not available');
+  }
+  if (!stack.updateSpec) {
+    stack.updateSpec = (newSpec, callback = () => {}) => {
+      Logger.error('Update Configuration not implemented in this browser');
+      callback('unimplemented');
+    };
+  }
+  if (!stack.setSignallingCallback) {
+    stack.setSignallingCallback = () => {
+      Logger.error('setSignallingCallback is not implemented in this stack');
+    };
+  }
+
+  that.close = () => {
+    Logger.debug('Closing ErizoConnection');
+    streamsMap.clear();
+    stack.close();
+  };
+
+  that.createOffer = () => {
+    stack.createOffer();
+  };
+
+  that.addStream = (stream) => {
+    Logger.debug(`message: Adding stream to Connection, streamId: ${stream.getID()}`);
+    streamsMap.add(stream.getID(), stream);
+    if (stream.local) {
+      stack.addStream(stream.stream);
+    }
+  };
+
+  that.removeStream = (stream) => {
+    const streamId = stream.getID();
+    if (!streamsMap.has(streamId)) {
+      Logger.warning(`message: Cannot remove stream not in map, streamId: ${streamId}`);
+      return;
+    }
+    streamsMap.remove(streamId);
+  };
+
+  that.processSignalingMessage = (msg) => {
+    stack.processSignalingMessage(msg);
+  };
+
+  that.sendSignalingMessage = (msg) => {
+    stack.sendSignalingMessage(msg);
+  };
+
+  that.createOffer = (isSubscribe) => {
+    stack.createOffer(isSubscribe);
+  };
+
+  that.enableSimulcast = (sdpInput) => {
+    stack.enableSimulcast(sdpInput);
+  };
+
+  that.updateSpec = (configInput, callback) => {
+    stack.updateSpec(configInput, callback);
+  };
+
+  // PeerConnection Events
+  if (stack.peerConnection) {
+    that.peerConnection = stack.peerConnection; // For backwards compatibility
+    stack.peerConnection.onaddstream = (stream) => {
+      if (that.onaddstream) {
+        Logger.error('OnAddStream from peerConnection - stream', stream);
+        that.onaddstream(stream);
+      }
+    };
+
+    stack.peerConnection.onremovestream = (stream) => {
+      if (that.onremovestream) {
```

Comment:
where can that.onremovestream be rewritten now?

Reply:
We're not doing it now, but decided to keep it there since the option existed in the original code.